File: Golang/Easy/383_Ransom_Note/solution.go
package ransomnote

func canConstruct(ransomNote string, magazine string) bool {
	magMap := make(map[string]int)

	for _, v := range magazine {
		magMap[string(v)] += 1
	}

	for _, v := range ransomNote {
		c := string(v)
		if _, exist := magMap[c]; !exist {
			return false
		}
		magMap[c] -= 1
		if magMap[c] < 0 {
			return false
		}
	}
	return true
}

File: Python/Easy/383_Ransom_Note/solution.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        map = {}
        for c in magazine:
            if c not in map:
                map[c] = 0
            map[c] += 1

        for c in ransomNote:
            if c not in map:
                return False
            map[c] -= 1
            if map[c] < 0:
                return False
        return True